File: flutter_inappwebview_windows/windows/types/server_trust_challenge.cpp
#include "../utils/flutter.h"
#include "server_trust_challenge.h"

namespace flutter_inappwebview_plugin
{
  ServerTrustChallenge::ServerTrustChallenge(const std::shared_ptr<URLProtectionSpace> protectionSpace)
    : URLAuthenticationChallenge(protectionSpace)
  {}

  flutter::EncodableMap ServerTrustChallenge::toEncodableMap() const
  {
    auto map = URLAuthenticationChallenge::toEncodableMap();
    return map;
  }
}

File: flutter_inappwebview_platform_interface/lib/src/types/server_trust_challenge.dart
import 'package:flutter_inappwebview_internal_annotations/flutter_inappwebview_internal_annotations.dart';

import 'url_authentication_challenge.dart';
import 'url_protection_space.dart';
import 'enum_method.dart';

part 'server_trust_challenge.g.dart';

///Class that represents the challenge of the [PlatformWebViewCreationParams.onReceivedServerTrustAuthRequest] event.
///It provides all the information about the challenge.
@ExchangeableObject()
class ServerTrustChallenge_ extends URLAuthenticationChallenge_ {
  ServerTrustChallenge_({required URLProtectionSpace_ protectionSpace})
      : super(protectionSpace: protectionSpace);
}
